solvation_analysis/plotting.py

```diff
@@ -0,0 +1,311 @@
+
+import plotly
+import plotly.graph_objects as go
+import plotly.express as px
+import matplotlib
+
+import numpy as np
+import pandas as pd
+
+
+def square_area(data, labels, cutoff=1):
+    data = np.floor(100 * data / np.sum(data)).astype(int)
+    do_exceed_cutoff = data > cutoff
+    label_array_short = np.repeat(labels[do_exceed_cutoff], data[do_exceed_cutoff])
+    # label_array = np.pad(
+    #     label_array_short,
+    #     (0, 100 - len(label_array_short)),
+    #     'constant',
+    #     constant_values=(np.nan, 'other'),
+    # )
+    colors = px.colors.qualitative.Dark24
+    color_array_short = np.repeat(colors[:sum(do_exceed_cutoff)], data[do_exceed_cutoff])
+    color_array = np.pad(
+        color_array_short,
+        (0, 100 - len(label_array_short)),
+        'constant',
+        constant_values=("000000", '#EEEEEE'),
+    )
+    rgb_array_unwrapped = [np.array(plotly.colors.hex_to_rgb(color)) for color in color_array]
+    rgb_values = np.reshape(rgb_array_unwrapped, (10,10,3))
+    fig = px.imshow(rgb_values, zmin=0, zmax=255)
+    fig.show()
+    return
+
+# single solution
+
+def plot_histogram(solution):
+    # histogram of what?
+    return
+
+def format_graph(fig, title, x_axis, y_axis):
```

Comment:
This feels a little bit unnecessary. I would just update these in the graph when required. :)

Reply:
What do you think of the idea of a general templating function? I am thinking each function could return a fairly bare bones plot that could then be prettied up with some formatting function.

```python
def format_plot(fig)
    # format fig
    return fig

fig = plot_something(data)
pretty_fig = format_plot(fig)

```

This might be overly complicated, but we could also make it a decorator.

```python
def format_plot(plotting_func):
    def pretty_plotting_func(data):
        fig = plotting_func(data)
        # format fig
        return fig
    return pretty_plotting_func
```

we could then add the formatting decorator to our definition of each plot. This would allow solvation_analysis to standardize a particular plotting style (I have become quite opinionated on the subject) with maximum code reuse.

```python
@pretty_plot
def format_plot(data)
    # make fig
    return fig 
```

Reply:
Sounds like a good idea to me!